xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp

```diff
@@ -163,12 +163,30 @@ void CEngineStats::SetDSP(bool state)
   m_hasDSP = state;
 }
 
+void CEngineStats::SetCurrentSinkFormat(AEAudioFormat &SinkFormat)
+{
+  CSingleLock lock(m_lock);
+  m_sinkFormat = SinkFormat;
+}
+
 bool CEngineStats::HasDSP()
 {
   CSingleLock lock(m_lock);
   return m_hasDSP;
 }
 
+bool CEngineStats::GetCurrentSinkFormat(AEAudioFormat &SinkFormat)
+{
+  CSingleLock lock(m_lock);
+  if(m_sinkFormat.m_dataFormat == AE_FMT_INVALID)
```

Comment:
Thats a good idea. I will update this, but the PR https://github.com/xbmc/xbmc/pull/7610 is also related on this new method.
If it is okay I will update both PRs.